packages/connection-manager/src/js/Connection/AbstractConnection.js

```diff
@@ -0,0 +1,91 @@
+import { EventEmitter } from "events";
+// import ConnectionEvent from "../ConnectionEvent/ConnectionEvent";
+
+/**
+ * AbstractConnection provides some default properties, methods, states and event-definitions used by Connection Manager
+ *
+ * Events which MUST be fired by all SubClasses:
+ * OPEN: when the connection actually blocks a pipe
+ * DATA: everytime data is recieved from the server
+ * CLOSE: when the connection frees its pipe
+ * ERROR: when an error occurs
+ */
+export default class AbstractConnection extends EventEmitter {
+  constructor(url) {
+    super();
+
+    if (this.constructor === AbstractConnection) {
+      throw new Error("Can't instantiate abstract class!");
+    }
+
+    if (!url) {
+      throw new Error("Can't instantiate without given URL!");
+    }
+    Object.defineProperty(this, "url", {
+      get: () => url
+    });
+
+    const created = Date.now();
+    Object.defineProperty(this, "created", {
+      get: () => created
+    });
+
+    let state = AbstractConnection.INIT;
+    Object.defineProperty(this, "state", {
+      get: () => state,
+      set: _state => {
+        if (
+          ![
+            AbstractConnection.INIT,
+            AbstractConnection.STARTED,
+            AbstractConnection.DONE,
+            AbstractConnection.CANCELLED
+          ].includes(_state)
+        ) {
+          throw new Error("Cant set Connection into unknown state.");
+        }
+        state = _state;
+      }
+    });
+
+    const symbol = Symbol("Connection:" + this.url + this.created);
+    Object.defineProperty(this, "symbol", {
+      get: () => symbol
+    });
+  }
+  static get INIT() {
+    return Symbol.for("DCOS.ConnectionManager.AbstractConnection.INIT");
```

Comment:
Let's define the symbols using constants and only pass them along instead of using `Symbol.for` to avoid recreation.